docs/guide/docs/main/overview.md

```diff
@@ -1,74 +1,109 @@
+
+While you can develop both kinds of bots with Botpress, you will benefit most if you're trying to build a smart bot.
 
 ## Anatomy of a bot
 
 A bot is pretty simple:
 
 1. It receives messages from Messaging **Channels**
 2. It **processes** these messages to understand, translate or escalate them
-3. It **decides** on what to respond back to the user
+3. It **decides** on what to respond to the user
 
 #### Lifecycle
 
 ![High-Level Diagram](assets/hld.png)
 
-<br/>
 
-In red are the developer extension points, i.e. where you will be able to inject code to customize the bot.
 
-## Channels
 
-A channel is a module that allows your bot to send and receive messages from a specific chat platform (Facebook Messenger, Telegram, etc.).
+The red dots are the developer extension points. These indicate where code can be injected to customize the bot.
+
+### Channels
 
-> Note: Unlike most other bot platforms, channels are installed and configured individually and locally, which means you have full control over the data that is transmitted between your bot and the chat platforms. Botpress does not proxy the messages to a 3rd party service as Microsoft Bot Framework does.
+A channel is a module that allows your bot to send and receive messages from a specific chat platform (Your website, Facebook Messenger, Telegram, etc.).
+
+> In Botpress, channels are installed and configured individually and locally, giving you full control over the data that is transmitted between your bot and the chat platforms. Botpress does not proxy the messages to a 3rd party service as Microsoft Bot Framework does.
 
 Behind the scenes, Botpress implements a queuing mechanism that processes ingoing and outgoing messages sequentially. If a message fails to be processed or sent for some reason, the message will be retried before raising an error to the developer and bot administrator.
 
-## NLU
+### Processing
+
+Messages received from messaging channels are then processed. During processing, Botpress extracts data from the text sent by a chat channel user. This data is processed in the following ways:
+
+#### NLU
 
 **Natural Language Understanding** (or **NLU**) involves your bot processing the messages received from the chat platforms, which are pure unstructured text, and transforming them into structured data that your bot will be able to work with. The main tasks the NLU engine does is:
 
-- **Intent Recognition**: recognizing what the user wants
-- **Entity Extraction**: extracting structured information from messages like dates, time, cities, names and more
-- **Slot tagging**: identify necessary **parameters** to fulfill given task
-- **Language Identification**: knowing in which language the user is writing
+- **Intent Recognition**: recognizing what the user wants.
```

Comment:
maybe add a word about OOS recognition?

Reply:
Good idea @franklevasseur can you draft a line for this. And replace OOS by Out of Sample or Out of Scope :)

Reply:
Something like this ?
```
**Out of Scope Recognition**: recognizing weither or not the user's intention can be handled by the bot.
**Out of Scope Recognition**: recognizing when the user wants something that can't be handled by the bot.
**Out of Scope Recognition**: identify when the user's intention falls outside of the bot competency.
```